netbox/ipam/forms/model_forms.py

```diff
@@ -341,6 +341,19 @@ def __init__(self, *args, **kwargs):
             self.fields['vminterface'].disabled = True
             self.fields['fhrpgroup'].disabled = True
 
+    # Correctly assigned assigned_object if the error exists when validating this form.
+    def add_error(self, field, errors):
```

Comment:
This isn't a tenable approach to fixing the validation. We need to adjust the logic under the model form's `clean()` method to ensure it's catching the error and raising a ValidationError against the appropriate field before the model's `full_clean()` method is called.

Reply:
> We need to adjust the logic under the model form's `clean()` method to ensure it's catching the error and raising a ValidationError against the appropriate field before the model's `full_clean()` method is called.

This isn't an option.

The "clean" flow is:

```
form.save() -> form.full_clean() -> form._clean_fields()
                                                       form._clean_form() -> form.clean()
                                                       form._post_clean() -> model.full_clean()
```

the form.clean_form() catches the validation error and adds it to the stack, however the post_clean calls the model clean regardless of whether there is already an error on the stack.

The error stack looks like this:

```
{
    "assigned_object": "Cannot reassign IP address while it is designated as the primary IP for the parent object",
    "vminterface": "Cannot reassign IP address while it is designated as the primary IP for the parent object"
}
```

The only way to remove this problem seems to be to also remove the ErrorDict from the model clean() and just throw a straight string validation error (pushing this change now).

I think realistically we need to re-investigate GFK's and see if there isn't a way to:

* Create a fieldgroup called "assigned_object" (or whatever the GFK name is)
* Create a select field called "assigned_object_type" (or whatever the GFK type field name is) that is populated with the types available for selection
* Create a select field called "assigned_object_id" (or whatever the GFK id field name is) that is populated with the objects available for selection based on the type selected
* Modify add_error to also match on the assigned_object fieldgroup

This obviously wouldn't be do-able in this FR but something to think about.